packages/sdk/wallets/wallet-base/src/wallet-base.ts

```diff
@@ -77,7 +77,8 @@ export abstract class WalletBase<TSigner extends Signer> implements ReadOnlyWall
       throw new Error('No transaction object given!')
     }
     const rlpEncoded = rlpEncodedTx(txParams)
-    const addToV = chainIdTransformationForSigning(txParams.chainId!)
+    const addToV =
+      rlpEncoded.type === 'celo-legacy' ? chainIdTransformationForSigning(txParams.chainId!) : 27
```

Comment:
nit: we could wrap 27 into a const as its kind of magic 🤷

Reply:
+1